twisted/names/test/test_rootresolve.py

```diff
@@ -209,20 +209,20 @@ def test_lookupChecksClass(self):
         badClass.CLASS = HS
         servers = {
             ('1.1.2.3', 53): {
-                ('foo.example.com', A): {
-                    'answers': [('foo.example.com', badClass)],
-                    'authority': [('foo.example.com', Record_NS('ns1.example.com'))],
-                    'additional': [('ns1.example.com', Record_A('10.0.0.2'))],
+                (b'foo.example.com', A): {
+                    'answers': [(b'foo.example.com', badClass)],
+                    'authority': [(b'foo.example.com', Record_NS(b'ns1.example.com'))],
+                    'additional': [(b'ns1.example.com', Record_A('10.0.0.2'))],
                 },
             },
             ('10.0.0.2', 53): {
-                ('foo.example.com', A): {
-                    'answers': [('foo.example.com', Record_A('10.0.0.3'))],
+                (b'foo.example.com', A): {
+                    'answers': [(b'foo.example.com', Record_A('10.0.0.3'))],
                 },
             },
         }
         resolver = self._getResolver(servers)
-        d = resolver.lookupAddress('foo.example.com')
+        d = resolver.lookupAddress(b'foo.example.com')
```

Comment:
I think this DNS class was originally implemented as per RFC 1035, which from my understanding, treats
names as a stream of ASCII characters:
https://tools.ietf.org/html/rfc1035#section-3.1

There are newer RFC's for dealing with "Internationalized Domain Names", and they specify a protocol
for encoding names:
https://tools.ietf.org/html/rfc5891#section-5
https://tools.ietf.org/html/rfc5895#section-2

using Unicode Normalization Form C (NFC)

I think any new `IResolverButGooder` should be implemented in terms of these newer RFCs.